src/server/call.ts

```diff
@@ -29,6 +29,19 @@ export class Call {
       throw new Error('Call info, method ID must be given');
     }
     let args: any = this.callInfo.bin_argument;
+    let metadata = new this.grpc.Metadata();
+    if (this.callInfo.metadata) {
+      for(let key in this.callInfo.metadata.map) {
+        // metadata is decoded into a funny structure, but we can access 
+        // what we need under the '.map' field
+        let metadataValues = this.callInfo.metadata.map[key].value.values;
+        if (metadataValues) {
+          for(let i=0; i < metadataValues.length; i++) {
```

Comment:
if we're keeping this overall code structure, this can still be simplified a bit: 
```
let metadataValues =  this.callInfo.metadata.map[key].value.values;
metadataValues.forEach(value => metadata.add(key, value));
```

Reply:
Ok, made this change.